rails/linter-config.md

```diff
@@ -0,0 +1,182 @@
+
+This Rake Task assumes that the gems for running the linters have been set up
+according to the documentation below.
+
+### Running Linters
+The Lint Rake file will provide the following Rake tasks:
+
+```bash
+# Runs all linters, configurable pass/fail in task code
+bundle exec rake lint
+# Linters individually, always return pass/fail exit codes
+bundle exec rake bundle:audit
+bundle exec rake rubocop
+bundle exec rake scss_lint
+bundle exec rake eslint:run
+```
+
+### `lint.rake`
+
+[ERB Templated `lint.rake` File](https://raw.githubusercontent.com/RoleModel/Rails-Template/master/templates/lib/tasks/lint.rake.tt)
+
+## Bundler Audit
+Bundler Audit reviews your Gemfile for any gems which need to be updated for
+security notices.
+
+* [Bundler-Audit Documentation](https://github.com/rubysec/bundler-audit)
+
+### Installing Bundler Audit
+```ruby
+group :development, :test do
+  gem 'bundler-audit'
+end
+```
+
+### Running Bundler Audit
+```bash
+bundle-audit check --update
+```
+
+## Rubocop
+Rubocop will determine if your code matches the Ruby style guide (or your
+project's configured adjustments).
+
+* [Rubocop Documentation](http://rubocop.readthedocs.io/en/latest/)
+* [Rubocop Shared Config](https://github.com/RoleModel/shared_rubocop)
+
+### Create the config file
+Create a `.rubocop.yml` config file in the root of your project:
+```yaml
+inherit_from:
+  - https://raw.githubusercontent.com/RoleModel/shared_rubocop/master/.rubocop.yml
+
+# This is a minimum Rubocop checks against, not necessarily what is actually
+#   used by the project.
+AllCops:
+  TargetRubyVersion: 2.2
+```
+
+When Rubocop is downloading the remote shared config file, it puts it in the
+project's root directory. We'll want to add it to `.gitignore`:
+
+```bash
+# Ignore Rubocop's local copy of our shared Rubocop config
+.rubocop-https*
+```
+
+### Install Rubocop
+Installing the gem under development and test environments allows your local,
+as well as Sempahore or other CI services, to be able to run Rubocop.
+```ruby
+group :development, :test do
+  gem 'rubocop', require: false
+end
+```
+And run `bundle` to install
+
+### Running Rubocop
+Assuming all config files are set up as specified above, and the gem is installed,
+running Rubocop in various forms can be done as follows:
+
+#### Local
+```bash
+bundle exec rubocop
+```
+
+#### Editor Plugins
+* [Atom](https://atom.io/packages/linter-rubocop)
+
+#### Semaphore / Other CI
+
+```bash
+bundle exec rubocop
+```
+
+You can also tell Semaphore to only exit with a failing error code on ERROR and
+above by specifying a `--fail-level` argument:
+
+```bash
+bundle exec rubocop --fail-level ERROR
+```
+
+## ESLint
+* [ESLint Documentation](http://eslint.org/docs/user-guide)
+
+### Create the config file
+You'll want to copy these two files into your project directory:
+* [.eslintrc.js](https://github.com/RoleModel/almanac/blob/add_rubocop_scss_and_eslint/.eslintrc.js)
```

Comment:
Whoops! Yes, the public copies should be in the Rails Template. I'll update the links shortly. (The files weren't in Rails Template when Ben and I were writing the first draft of this document.)

Thanks for pointing this out. :)